### PR TITLE
Fixes #14258: check-rudder-agent runs every  5 minutes exactly by cron, and can cause spike in resource usage

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/14258

### CODE REVIEW

File: share/commands/agent-check [hunk 1]
```diff
@@ -7,25 +7,32 @@
 # @man - clean lock file if it is too big
 # @man - check and restore init files 
 # @man - check that promises have been properly copied
+# @man - will sleep a random time (max half the run interval) when not run in interactive mode priori checking
```
Comment (peckpeck) [inline share/commands/agent-check]: priori checking ?

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, squash merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**